### PR TITLE
[MRG] MAINT: pin numpy version in windows

### PR DESCRIPTION
**Pull request by glemaitre**
Trying to lower the NumPy version when building the wheels.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -79,6 +85,7 @@ install:
   - "git reset --hard"
 
   # Install the build and runtime dependencies of the project.
+  - "%CMD_IN_ENV% pip install --timeout=60 numpy==%NP_BUILD_DEP%"
```
[the remark this answers is not in the document]
Comment (glemaitre) [reply appveyor.yml]: All the minimum version are taken from PyPi now. I got issue with the previous built `1.9.3` from the rackspace.

[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply appveyor.yml]: Ah, yes, I see that pypi has windows wheels starting at 1.10

[the remark this answers is not in the document]
Comment (glemaitre) [reply appveyor.yml]: However, there is a failing test in Python 3.5 which seems weird.

[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply appveyor.yml]: I suppose you can then actually remove the rackspace host below? as it is not needed anymore for installing the remaining of requirements.txt ?

[the remark this answers is not in the document]
Comment (ogrisel) [reply appveyor.yml]: +1 for removing the rackspace config if it's no longer used.

**Comment by ogrisel** [comment]
The failure on Python 3.5 is weird. Maybe you can try with numpy 1.12.1 for Python 3.5 so see it this is a bug in numpy?

**Comment by jorisvandenbossche** [comment]
If that is the case, I would still build with numpy 1.10, but test with numpy 1.12. In the travis configuration a similar distinction is used.

**Comment by ogrisel** [comment]
> If that is the case, I would still build with numpy 1.10, but test with numpy 1.12. In the travis configuration a similar distinction is used.

I agree.

**Comment by glemaitre** [comment]
I am doing a build bumping numpy. I will try to investigate the failure in a virtual machine.

**Comment by ogrisel** [comment]
Thanks!

**Comment by glemaitre** [comment]
So the reason is that `np.clip` does not discard NaN in this version

```python
In [1]: import numpy as np

In [2]: X = np.array([np.nan, 1, 2, np.nan, 10])

In [3]: np.clip(X, 0, 5)
Out[3]: array([ 0.,  1.,  2.,  0.,  5.])
```

in the other version, we get something like:

```python
array([np.nan, 1., 2., np.nan, 5.])
```

I'll try to check why.

**Comment by glemaitre** [comment]
I am testing with an earlier version. The bug is actually this one:
https://github.com/numpy/numpy/issues/7601

**Comment by glemaitre** [comment]
I don't have strong opinion but it could be the most common case for most users.

**Comment by ogrisel** [comment]
Thanks for the fix!

**Pull request merged**